### PR TITLE
Apply and enforce formatting

### LINKED ISSUE #152: Rustfmt is not being applied/checked
i've submitted a couple of pull requests now and i notice that rustfmt hasn't been applied to this crate. That is, if i run rustfmt according to the 'contributing' guide, then it makes changes to the formatting.

while not a big deal, it could lead to a few mildly painful rebases. It's also really easy to catch.

I'm happy to submit a pull request for this?

could catch it using a github action, and/or a pre-commit hook.

thoughts?

### PR DESCRIPTION
**Pull request by danieleades**
apply formatting, and add a CI action to check formatting

closes #152

### CODE REVIEW

**Comment by danieleades** [comment]
rebased on master

**Comment by danieleades** [comment]
rebase and tidy commits

**Comment by danieleades** [comment]
github action run results can be checked here - https://github.com/danieleades/upgrade/pull/3/checks?check_run_id=2360001747

File: src/release/check.rs [hunk 1]
```diff
@@ -125,12 +125,12 @@ fn next_(
             next = "21.04";
 
             ReleaseStatus {
-                build:   if development { release_check(next) } else { BuildStatus::Blacklisted },
+                build: if development { release_check(next) } else { BuildStatus::Blacklisted },
                 current: "20.10",
-                is_lts:  false,
+                is_lts: false,
```
Comment (crawfxrd) [inline src/release/check.rs:130]: These don't seem right: it's *removing* the field alignment, whereas everything else *adds* field alignment. What's causing fmt to do that?

Comment (danieleades) [reply src/release/check.rs:130]: Good catch. Rustfmt bug? Will do some digging

Comment (danieleades) [reply src/release/check.rs:130]: if the difference in field widths is too great, this formatting rule doesn't apply. this threshold is configured to 30 characters. the max diff of field widths in this struct is greater than 30, so this is correct

the rule itself is unstable

**Review by bflanagin — APPROVED** [review]
No regressions detected.

**Review by crawfxrd — APPROVED** [review]

**Comment by mmstick** [comment]
Formatting changes will destroy mergeability of development branch. Had to undo the recently-merged commits.

**Comment by danieleades** [comment]
@mmstick I take it you don't want to persue enforcing the formatting policy in CI?

**Comment by mmstick** [comment]
Maybe when it's not actively in development

**Comment by danieleades** [comment]
> Maybe when it's not actively in development

Fair enough. Though I would point out that a pull request to add formatting is, by it's very nature, very easy to rebase.

**Pull request closed without merging**